crawl-ref/source/dat/des/altar/overflow.des

```diff
@@ -88,6 +88,25 @@ function callback.hep_ancestral_shrine_los(data, triggerable, triggerer, marker,
   end
 end
 
+function callback.xom_greatest_gift(data, triggerable, triggerer, marker, ev)
+  if not dgn.persist.xom_gg_heard and you.god() == "Xom" then
+    dgn.persist.xom_gg_heard = true
+    dgn.persist.xom_gg_granted = true
+  elseif not dgn.persist.xom_gg_heard and you.god() ~= "Xom" then
+    crawl.god_speaks("Xom", '"WORSHIP ME, MORTAL," Xom booms, "AND I SHALL'
+                            .. ' GRANT YOU THE GREATEST GIFT OF ALL."')
+    dgn.persist.xom_gg_heard = true
+  elseif not dgn.persist.xom_gg_granted and you.god() == "Xom" then
+    local pos = dgn.find_marker_positions_by_prop("gift", "altar")
+    dgn.create_monster(pos[1].x, pos[1].y, "generate_awake att:friendly goblin")
```

Comment:
I think it'd be better for this to be att:good_neutral; I'd rather not let the player have a real ally that they can order around, cart between levels, etc.

Also, it is very funny if the player decides to instantly murder the goblin to avoid losing XP from things it kills. Friendship!